File: app/src/main/java/com/aaron/yespdf/main/AllFragment.java
package com.aaron.yespdf.main;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.aaron.base.base.BaseFragment;
import com.aaron.yespdf.R;
import com.aaron.yespdf.R2;
import com.aaron.yespdf.common.CommonFragment;
import com.aaron.yespdf.common.DBHelper;
import com.aaron.yespdf.common.DataManager;
import com.aaron.yespdf.common.UiManager;
import com.aaron.yespdf.common.XGridDecoration;
import com.aaron.yespdf.common.YGridDecoration;
import com.aaron.yespdf.common.bean.Cover;
import com.aaron.yespdf.common.event.AllEvent;
import com.blankj.utilcode.util.LogUtils;
import com.blankj.utilcode.util.ThreadUtils;

import org.greenrobot.eventbus.EventBus;
import org.greenrobot.eventbus.Subscribe;
import org.greenrobot.eventbus.ThreadMode;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.Unbinder;

/**
 * @author Aaron aaronzzxup@gmail.com
 */
public class AllFragment extends CommonFragment implements IOperation, AbstractAdapter.ICommInterface<Cover> {

    @BindView(R2.id.app_rv_all)
    RecyclerView rvAll;

    private Unbinder unbinder;
    private AllAdapter adapter;

    private List<Cover> coverList = DataManager.getCoverList();
    private List<Cover> selectCollections = new ArrayList<>();

    static Fragment newInstance() {
        return new AllFragment();
    }

    public AllFragment() {

    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        EventBus.getDefault().register(this);
        View layout = inflater.inflate(R.layout.app_fragment_all, container, false);
        unbinder = ButterKnife.bind(this, layout);
        initView();
        return layout;
    }

    @Override
    public void onResume() {
        super.onResume();
        ((MainActivity) activity).injectOperation(this);
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        unbinder.unbind();
        EventBus.getDefault().unregister(this);
    }

    @Subscribe(threadMode = ThreadMode.MAIN)
    public void onPdfDeleteEvent(PdfDeleteEvent event) {
        LogUtils.e(event);
        if (event.isEmpty) {
            DBHelper.deleteCollection(event.dir);
        }
        update();
    }

    @Subscribe(threadMode = ThreadMode.MAIN)
    public void onAllEvent(AllEvent event) {
        if (event.isEmpty) {
            DBHelper.deleteCollection(event.dir);
        }
        update();
    }

    @Override
    public void onStartOperation() {
        ((MainActivity) activity).startOperation();
    }

    @Override
    public void onSelect(List<Cover> list, boolean selectAll) {
        selectCollections.clear();
        selectCollections.addAll(list);
        ((MainActivity) activity).selectResult(list.size(), selectAll);
    }

    @Override
    public void delete() {
        if (!selectCollections.isEmpty()) {
            ThreadUtils.executeByIo(new ThreadUtils.SimpleTask<List<String>>() {
                @Override
                public List<String> doInBackground() {
//                    coverList.removeAll(selectCollections);
                    List<String> list = DBHelper.deleteCollection(selectCollections);
                    DataManager.updateAll();
                    return list;
                }

                @Override
                public void onSuccess(List<String> dirList) {
                    UiManager.showShort(R.string.app_delete_completed);
                    ((MainActivity) activity).finishOperation();
                    adapter.notifyDataSetChanged();
                    EventBus.getDefault().post(new AllDeleteEvent(dirList));
                }
            });
        }
    }

    @Override
    public void selectAll(boolean selectAll) {
        adapter.selectAll(selectAll);
    }

    @Override
    public void cancelSelect() {
        adapter.cancelSelect();
    }

    @Override
    public String deleteDescription() {
        return getString(R.string.app_will_delete) + " " + selectCollections.size() + " " + getString(R.string.app_delete_for_all);
    }

    void update() {
        DataManager.updateCollection();
        adapter.reset();
        adapter.notifyDataSetChanged();
    }

    private void initView() {
        rvAll.addItemDecoration(new XGridDecoration());
        rvAll.addItemDecoration(new YGridDecoration());
        GridLayoutManager lm = new GridLayoutManager(activity, 3);
        lm.setSpanSizeLookup(new GridLayoutManager.SpanSizeLookup() {
            @Override
            public int getSpanSize(int position) {
                if (coverList.isEmpty()) {
                    return 3;
                }
                return 1;
            }
        });
        rvAll.setLayoutManager(lm);
        adapter = new AllAdapter(this, getFragmentManager(), coverList);
        rvAll.setAdapter(adapter);
    }
}

File: app/src/main/java/com/aaron/yespdf/main/AllFragment.kt
package com.aaron.yespdf.main

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.GridLayoutManager.SpanSizeLookup
import com.aaron.yespdf.R
import com.aaron.yespdf.common.*
import com.aaron.yespdf.common.bean.Cover
import com.aaron.yespdf.common.event.AllEvent
import com.aaron.yespdf.main.AbstractAdapter.ICommInterface
import com.blankj.utilcode.util.LogUtils
import com.blankj.utilcode.util.ThreadUtils
import com.blankj.utilcode.util.ThreadUtils.SimpleTask
import kotlinx.android.synthetic.main.app_fragment_all.*
import org.greenrobot.eventbus.EventBus
import org.greenrobot.eventbus.Subscribe
import org.greenrobot.eventbus.ThreadMode
import java.util.*

/**
 * @author Aaron aaronzzxup@gmail.com
 */
class AllFragment : CommonFragment(), IOperation, ICommInterface<Cover> {

    private var adapter: AllAdapter? = null
    private val coverList = DataManager.getCoverList()
    private val selectCollections: MutableList<Cover> = ArrayList()

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        EventBus.getDefault().register(this)
        return inflater.inflate(R.layout.app_fragment_all, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        initView()
    }

    override fun onResume() {
        super.onResume()
        (activity as MainActivity).injectOperation(this)
    }

    override fun onDestroyView() {
        super.onDestroyView()
        EventBus.getDefault().unregister(this)
    }

    @Subscribe(threadMode = ThreadMode.MAIN)
    fun onPdfDeleteEvent(event: PdfDeleteEvent) {
        LogUtils.e(event)
        if (event.isEmpty) {
            DBHelper.deleteCollection(event.dir)
        }
        update()
    }

    @Subscribe(threadMode = ThreadMode.MAIN)
    fun onAllEvent(event: AllEvent) {
        if (event.isEmpty) {
            DBHelper.deleteCollection(event.dir)
        }
        update()
    }

    override fun onStartOperation() {
        (activity as MainActivity).startOperation()
    }

    override fun onSelect(list: List<Cover>, selectAll: Boolean) {
        selectCollections.clear()
        selectCollections.addAll(list)
        (activity as MainActivity).selectResult(list.size, selectAll)
    }

    override fun delete() {
        if (selectCollections.isNotEmpty()) {
            ThreadUtils.executeByIo<List<String>>(object : SimpleTask<List<String>>() {
                override fun doInBackground(): List<String>? {
                    val list = DBHelper.deleteCollection(selectCollections)
                    DataManager.updateAll()
                    return list
                }

                override fun onSuccess(dirList: List<String>?) {
                    UiManager.showShort(R.string.app_delete_completed)
                    (activity as MainActivity).finishOperation()
                    adapter?.notifyDataSetChanged()
                    dirList?.run { EventBus.getDefault().post(AllDeleteEvent(this)) }
                }
            })
        }
    }

    override fun selectAll(selectAll: Boolean) {
        adapter?.selectAll(selectAll)
    }

    override fun cancelSelect() {
        adapter?.cancelSelect()
    }

    override fun deleteDescription(): String? {
        return getString(R.string.app_will_delete) + " " + selectCollections.size + " " + getString(R.string.app_delete_for_all)
    }

    fun update() {
        DataManager.updateCollection()
        adapter?.reset()
        adapter?.notifyDataSetChanged()
    }

    private fun initView() {
        app_rv_all.addItemDecoration(XGridDecoration())
        app_rv_all.addItemDecoration(YGridDecoration())
        val lm = GridLayoutManager(activity, 3)
        lm.spanSizeLookup = object : SpanSizeLookup() {
            override fun getSpanSize(position: Int): Int {
                return if (coverList.isEmpty()) {
                    3
                } else 1
            }
        }
        app_rv_all.layoutManager = lm
        adapter = fragmentManager?.let { AllAdapter(this, it, coverList) }
        app_rv_all.adapter = adapter
    }

    companion object {
        fun newInstance(): Fragment {
            return AllFragment()
        }
    }
}